### DISCUSSION TITLE
Set static ip on ethernet

### DISCUSSION THREAD

**Discussion by apple2ms** [tesla-android/issue-tracker#223] [General]
Hi, 
Is there an easy way to set a static ip for the wired ethernet?
I am hoping to set it at 192.168.90.120 and mask 255.255.255.0 and connect to the car using the diag port. I understand it is filtering all traffic except on port 22 but I can use ssh reverse tunnel to get port 80 on the tesla mcu. 
Basically If I could set the ip I can get it all working through cable on a rooted tesla.

**Comment by mikegapinski** [comment]
I will let you know if I can get it working. I have a rooted mcu2 to test on.

Any idea if there is any magic trick we could potentially do to get this working for not rooted cars also? 

If you want the best performance you can connect a Wi-Fi antenna directly to the MCU on a very short adapter cable. It's even faster than Ethernet, older units don't have a gigabit port to begin with.

**Reply by apple2ms** [reply]
Not much can be done on non-rooted cars because we can't reconfigure ssh and do reverse tunneling. The diag port is very restricted due to the settings of the ethernet switch inside the mcu. I guess what could be done is to connect through the autopilot port which would work but then the ap won't work obviously.. and that needs an ethernet to two-wire-ethernet which is probably not too fast anyway.
Can I put the static ip in a file somewhere or maybe use some app to set it? Just for a quick test..
It would be great if we can get the mcu to share its lte to the pi.

**Comment by apple2ms** [comment]
I have managed to get it to show up on the tesla browser through the
diagnostic port.
Added a router to dhcp 192.168.90.x addresses and connected the tesla diag
port and rpi to it.
using connectbot on pi I connect to the tesla and do remote port forwarding
settings for ports 80 and 8080.
I also need to drop 2 iptables rules on the tesla mcu:
/usr/sbin/iptables -D  INTERNET -d 127.0.0.0/8 -j REJECT --reject-with
icmp-port-unreachable;
/usr/sbin/iptables -D  INTERNET -o lo -j REJECT --reject-with
icmp-port-unreachable;
so that I can open localhost on the web browser.
Unfortunately due to ssh encryption the link is too slow..
We need another way to do port forwarding through port 22.
Something through socat maybe?
